### PR TITLE
Rename controller names

### PR DESCRIPTION
**Pull request by omegas27**
Towards https://github.com/giantswarm/giantswarm/issues/12550

I followed the practice defined https://github.com/giantswarm/aws-operator/blob/02d8199912361284bffde9d6d034998baa6fe5ba/service/controller/control_plane.go#L115

### CODE REVIEW

**Comment by omegas27** [comment]
Tested on godsmack

File: service/controller/clusterapi/controller.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ func NewController(config ControllerConfig) (*Controller, error) {
 		c := controller.Config{
 			K8sClient: config.K8sClient,
 			Logger:    config.Logger,
-			Name:      "clusterapi-controller",
+			Name:      project.Name() + "-cluster-api-controller",
```
Comment (JosephSalisbury) [inline service/controller/clusterapi/controller.go:49]: Do we want to specify if this is for CPs or TCs?

Comment (omegas27) [reply service/controller/clusterapi/controller.go:49]: I was thinking we would actually have a control-plane controller as we do not need to differentiate between cluster api and legacy there

Comment (omegas27) [reply service/controller/clusterapi/controller.go:49]: But we can specify this is for tenant clusters if we think this is necessary

Comment (JosephSalisbury) [reply service/controller/clusterapi/controller.go:49]: I think this is fine, we'll know which one is which, ez

**Review by JosephSalisbury — APPROVED** [review]
lgmt if @TheoBrigitte agrees

**Comment by JosephSalisbury** [comment]
lgmt? looks great mo tee?

**Review by TheoBrigitte — APPROVED** [review]
Fine with me, let's just find an efficient way to remove those finalisers

**Comment by omegas27** [comment]
@JosephSalisbury can we write a batch script that patches finalizers to removes only some of them on all awsconfig, azureconfigs, kvmconfigs and clusters CR? 
I would image we could extract the list of finalizers of a CR, remove the ones we know and then reapply it?

**Comment by TheoBrigitte** [comment]
I also thought about using patches but, there's no way to remove an element from an array by value. Doing as you suggest has a risk to miss some finalisers added/removed in between.

**Comment by omegas27** [comment]
Then manually editing it is the only way? 
I found out that we can edit all the awsconfig in directly using `kubectl edit awsconfigs` and not do them one by one though

**Comment by TheoBrigitte** [comment]
Then do a quick find & delete and we're done. Let's do that.

**Pull request merged**